plugins/outputs/newrelic/newrelic_test.go

```diff
@@ -0,0 +1,134 @@
+package newrelic
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/newrelic/newrelic-telemetry-sdk-go/telemetry"
+	"github.com/stretchr/testify/require"
+)
+
+func ErrorContains(out error, want string) bool {
+	if out == nil {
+		return want == ""
+	}
+	if want == "" {
+		return false
+	}
+	return strings.Contains(out.Error(), want)
+}
+
+func TestBasic(t *testing.T) {
+	nr := &NewRelic{
+		MetricPrefix: "Test",
+		InsightsKey:  "12345",
+		Timeout:      internal.Duration{Duration: time.Second * 5},
+	}
+
+	err := nr.Connect()
+	require.NoError(t, err)
+
+	err = nr.Write(testutil.MockMetrics())
+	if !ErrorContains(err, "unable to harvest metrics ") {
+		t.Errorf("unexpected error: %v", err)
+	}
+}
+
+func TestNewRelic_Write(t *testing.T) {
+	type fields struct {
+		harvestor    *telemetry.Harvester
+		InsightsKey  string
+		MetricPrefix string
+		Timeout      internal.Duration
+	}
+	type args struct {
+		metrics []telegraf.Metric
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		metrics []telegraf.Metric
+		wantErr bool
+	}{
+		{
+			name: "Test: Basic mock metric write",
+			fields: fields{
+				InsightsKey:  "insightskey",
+				MetricPrefix: "test1",
+				Timeout:      internal.Duration{Duration: time.Second * 15},
+			},
+			metrics: testutil.MockMetrics(),
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nr := &NewRelic{
+				harvestor:    tt.fields.harvestor,
+				InsightsKey:  tt.fields.InsightsKey,
+				MetricPrefix: tt.fields.MetricPrefix,
+			}
+			if err := nr.Write(tt.metrics); (err != nil) != tt.wantErr {
+				t.Errorf("NewRelic.Write() error = %v, wantErr %v", err, tt.wantErr)
+			}
```

Comment:
It seems that this test doesn't have any proper assertions, the only way it can fail is if Write panics.  Each test should exercise a specific error condition or verify a particular behavior.

Reply:
Rewrote this test case to validate the metrics string that is posted to New relic backend. The test now test various datatypes that can be present in the telegraf metrics type